Conventional/Conventions/Cecil/MustNotResolveCurrentTimeViaDateTimeDirectlyConventionSpecification.cs

```diff
@@ -1,38 +1,46 @@
-﻿using System;
-using System.Linq;
-using Conventional.Conventions;
-using Mono.Cecil;
-using Mono.Cecil.Cil;
-
-namespace Conventional.Cecil.Conventions
-{
-    public class MustNotUseDateTimeNowConventionSpecification : ConventionSpecification
-    {
-        protected override string FailureMessage
-        {
-            get { return "Must not use DateTime.Now. It is called {0} times in this type."; }
-        }
-
-        public override ConventionResult IsSatisfiedBy(Type type)
-        {
-            var nowAssignments =
-                type.ToTypeDefinition()
-                    .Methods
-                    .Where(method => method.HasBody)
-                    .SelectMany(method => method.Body.Instructions)
-                    .Where(
-                        x =>
-                            x.OpCode == OpCodes.Call && x.Operand is MethodReference &&
-                            ((MethodReference) x.Operand).DeclaringType.FullName == "System.DateTime" &&
-                            ((MethodReference) x.Operand).Name == "get_Now")
-                    .ToArray();
-
-            if (nowAssignments.Any())
-            {
-                return ConventionResult.NotSatisfied(type.FullName, FailureMessage.FormatWith(nowAssignments.Count()));
-            }
-
-            return ConventionResult.Satisfied(type.FullName);
-        }
-    }
+﻿using System;
+using System.Linq;
+using Conventional.Conventions;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace Conventional.Cecil.Conventions
+{
+    public class MustNotResolveCurrentTimeViaDateTimeDirectlyConventionSpecification : ConventionSpecification
+    {
+        protected override string FailureMessage
+        {
```

Comment:
Good catch! Do you want to get rid of "It is called {0} times in this type."? It seems that piece is used only in here and for DataTimeOffset convention. Given we are checking not only against "Now" now, it should be transformed a bit into "Those are called {0} times in this type". I wonder if this is really that useful esp. given that would now be aggregate sum for all calls we are catching. If it is to be kept though, I would suggest to make "this" from "this type" actual type name.